internal/sysfs/osfile.go

```diff
@@ -86,6 +86,9 @@ func (f *osFile) SetAppend(enable bool) (errno experimentalsys.Errno) {
 	// Clear any create flag, as we are re-opening, not re-creating.
 	f.flag &= ^experimentalsys.O_CREAT
 
+	// Clear trunc flag, as we are re-opening.
+	f.flag &= ^experimentalsys.O_TRUNC
```

Comment:
Since this comment repeats what's above, how about merging the two?
```suggestion
	f.flag &= ^(experimentalsys.O_CREAT | experimental.O_TRUNC)
```